### PR TITLE
fix: add 3.11 to classifiers and upgrade Docker

### PR DESCRIPTION
**Pull request by sriram-mv**
- update dependencies, need to nail down the versions.

#### Mandatory Checklist
**PRs will only be reviewed after checklist is complete**

- [ ] Add input/output [type hints](https://docs.python.org/3/library/typing.html) to new functions/methods
- [ ] Write design document if needed ([Do I need to write a design document?](https://github.com/aws/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.md#design-document))
- [ ] Write/update unit tests
- [ ] Write/update integration tests
- [ ] Write/update functional tests if needed
- [ ] `make pr` passes
- [ ] `make update-reproducible-reqs` if dependencies were changed
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

File: samcli/local/docker/container.py [hunk 1]
```diff
@@ -206,9 +206,6 @@ def create(self):
             # Ex: 128m => 128MB
             kwargs["mem_limit"] = "{}m".format(self._memory_limit_mb)
 
-        if self.network_id == "host":
```
Comment (sriram-mv) [inline samcli/local/docker/container.py:209]: Removing this as this never worked anyways.

File: requirements/dev.txt [hunk 2]
```diff
@@ -16,14 +16,14 @@ types-psutil==5.9.5.12
...
-pytest==7.2.2
-parameterized==0.9.0
-pytest-xdist==3.2.0
-pytest-forked==1.6.0
-pytest-timeout==2.1.0
-pytest-rerunfailures==11.1.2
-pytest-json-report==1.5.0
-filelock==3.12.0
+pytest~=7.2.2
+parameterized
```
Comment (sriram-mv) [inline requirements/dev.txt:20]: Deps needs to be locked down.

File: requirements/dev.txt [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
-pytest-cov==4.0.0
+coverage
+pytest-cov
 
 
 # type checking and related stubs
 # mypy adds new rules in new minor versions, which could cause our PR check to fail
 # here we fix its version and upgrade it manually in the future
-mypy==0.790
+mypy
```
Comment (sriram-mv) [inline requirements/dev.txt:10]: Deps need to be locked down.

File: requirements/base.txt [hunk 4]
```diff
@@ -3,21 +3,21 @@ click~=8.0
...
-jmespath~=0.10.0
-ruamel_yaml==0.17.21
+jmespath~=1.0.1
+ruamel_yaml~=0.17.21
 PyYAML>=5.4.1,==5.*
 cookiecutter~=2.1.1
 aws-sam-translator==1.68.0
 #docker minor version updates can include breaking changes. Auto update micro version only.
-docker~=4.2.0
+docker~=6.1.0
```
Comment (sriram-mv) [inline requirements/base.txt:12]: Integ tests seem to be passing.

File: samcli/lib/build/workflows.py [hunk 5]
```diff
@@ -4,7 +4,7 @@
 from typing import List
 
 CONFIG = namedtuple(
-    "Capability",
+    "CONFIG",
```
Comment (sriram-mv) [inline samcli/lib/build/workflows.py:7]: is this right?

File: requirements/dev.txt [hunk 6]
```diff
@@ -7,16 +7,24 @@ pytest-cov==4.0.0
...
 # here we fix its version and upgrade it manually in the future
-mypy==0.790
+mypy==1.3.0
 boto3-stubs[apigateway,cloudformation,ecr,iam,lambda,s3,schemas,secretsmanager,signer,stepfunctions,sts,xray]==1.26.131
 types-pywin32==306.0.0.0
 types-PyYAML==6.0.12
 types-chevron==0.14.2.4
 types-psutil==5.9.5.12
 types-setuptools==65.4.0.0
+types-Pygments==2.15.0.1
```
Comment (sriram-mv) [inline requirements/dev.txt:17]: thanks for adding the others!

File: samcli/local/docker/container.py (hunk 1 above)
Comment (sriram-mv) [reply samcli/local/docker/container.py:209]: Should we add a help text or note. along with a pinned issue for this?

File: samcli/lib/build/workflows.py (hunk 5 above)
Comment (jfuss) [reply samcli/lib/build/workflows.py:7]: I could ignore them but mypy complains. This is the "name-match" error mypy throws.

File: samcli/cli/hidden_imports.py [hunk 7]
```diff
@@ -2,31 +2,23 @@
...
+def walk_modules(module: ModuleType, visited: set) -> None:
     """Recursively find all modules from a parent module"""
     for pkg in pkgutil.walk_packages(module.__path__, module.__name__ + "."):
         if pkg.name in visited:
             continue
         visited.add(pkg.name)
         if pkg.ispkg:
             submodule = __import__(pkg.name)
-            submodule = cast(HasPathAndName, submodule)
+            submodule = submodule
```
Comment (hawflau) [inline samcli/cli/hidden_imports.py:16]: should we just remove this line?

Comment (jfuss) [reply samcli/cli/hidden_imports.py:16]: Done

**Review by hawflau — APPROVED** [review]

**Review by mildaniel — APPROVED** [review]

**Pull request merged**